app/react/src/client/preview/utils.ts

```diff
@@ -0,0 +1,114 @@
+import { defaultDecorateStory, combineParameters } from '@storybook/client-api';
+import { GlobalConfig, Meta, Story, StoryContext } from './types-6-0';
+
+/**
+ * Function that will receive a story along with meta (e.g. a default export from a .stories file)
+ * and optionally a globalConfig e.g. (import * from '../.storybook/preview)
+ * and will return a composed component that has all args/parameters/decorators/etc combined and applied to it.
+ *
+ *
+ * It's very useful for reusing a story in scenarios outside of Storybook like unit testing.
+ *
+ * Example:
+ *```jsx
+ * import { render } from '@testing-library/react';
+ * import { composeStory } from '@storybook/react';
+ * import Meta, { Primary as PrimaryStory } from './Button.stories';
+ * import * as globalConfig from '../.storybook/preview';
+ *
+ * const Primary = composeStory(PrimaryStory, Meta, globalConfig);
+ *
+ * test('renders primary button with Hello World', () => {
+ *   const { getByText } = render(<Primary>Hello world</Primary>);
+ *   expect(getByText(/Hello world/i)).not.toBeNull();
+ * });
+ *```
+ *
+ * @param story
+ * @param meta - e.g. (import Meta from './Button.stories')
+ * @param [globalConfig] - e.g. (import * as globalConfig from '../.storybook/preview')
+ */
+export function composeStory<GenericArgs>(
+  story: Story<GenericArgs>,
+  meta: Meta,
+  globalConfig: GlobalConfig = {}
+) {
+  const finalStoryFn = (context: StoryContext) => {
+    const { passArgsFirst = true } = context.parameters;
+    if (!passArgsFirst) {
+      throw new Error(
+        'composeStory does not support legacy style stories (with passArgsFirst = false).'
+      );
+    }
+    return story(context.args as GenericArgs, context);
```

Comment:
```suggestion
    return passArgsFirst ? story(context.args as GenericArgs, context) : story(context);
```

Was there a reason not to just do this? I'm not sure if it is a big deal to support non-`passArgsFirst` but this is simple enough?

Reply:
IMHO we shouldn't support legacy CSF on new features. if you want to use it, migrate!

Reply:
Let me know if we should resolve this conversation :D
I'm happy to do either way